rslib/src/notes/undo.rs

```diff
@@ -64,20 +67,47 @@ impl Collection {
         if changes.op != Op::UpdateNote {
             return;
         }
-
-        if let Some(previous_op) = self.previous_undo_op() {
-            if previous_op.kind != Op::UpdateNote {
-                return;
+        let Some(previous_op) = self.previous_undo_op() else {
+            return;
+        };
+        if previous_op.kind != Op::UpdateNote {
+            return;
+        }
+        let Some(current_op) = self.current_undo_op() else {
+            return;
+        };
+        let is_col_modified_change = |change: &&UndoableChange| match change {
+            UndoableChange::Collection(col_change) => {
+                matches!(col_change, UndoableCollectionChange::Modified(_))
             }
-
-            if let (
-                Some(UndoableChange::Note(UndoableNoteChange::Updated(previous))),
-                Some(UndoableChange::Note(UndoableNoteChange::Updated(current))),
-            ) = (
-                previous_op.changes.last(),
-                self.current_undo_op().and_then(|op| op.changes.last()),
-            ) {
-                if previous.id == current.id && previous_op.timestamp.elapsed_secs() < 60 {
+            _ => false,
+        };
+
+        let changes_to_pop = current_op
+            .changes
+            .iter()
+            .rev()
+            .take_while(is_col_modified_change)
+            .collect_vec()
+            .len()
+            + 1;
+        let current_op_change = current_op
+            .changes
+            .iter()
+            .rev()
+            .find(|c| !is_col_modified_change(c));
```

Comment:
While very unlikely, this could theoretically drop the entire undo entry if the operation changed multiple notes, and the last change of the two operations was for the same note. Perhaps it would make more sense to compare .first() instead? This coalescing is only intended to happen in the single-note update case.

Reply:
Hmm, will it drop the entire entry though? The code is only popping UndoableCollectionChange::Modified changes and a single UndoableNoteChange::Updated at the end of the vec. I changed that anyway.

Reply:
Sorry, you're right - I misread the code. Thinking about it more, it might make the code a bit simpler to explicitly match/clear the entire list of changes. I've pushed a tweak that does this - does it look ok to you? Apologies for the changing requirements :-)

Reply:
Looking better to me!